### DISCUSSION TITLE
Project HÆLTH Blog #3

### DISCUSSION THREAD

**Discussion by OmarMoujahed** [ldcdorn/haelth#44] [General]
# Project HÆLTH 🌱
Our third blog post covers the topics we worked on and goals we achieved this week.

## Activity Diagrams
While working on the activity diagrams, we found that [draw.io](https://app.diagrams.net/) is a great tool for creating them, especially for simultaneous collaboration. We created three activity diagrams illustrating the functionalities of each of the three categories in the application: Workout, Nutrition, and Cognitive Exercise.

![screenshot_aktivitatsdiagramm_workout](https://github.com/user-attachments/assets/cc01fed6-ab59-41df-ac98-3959465a132c)

For example, this activity diagram visualizes the application's workout functionalities, such as saving completed exercises. To view all activity diagrams and the requirements they address, we recommend referring to the [Software Requirements Specification](https://github.com/ldcdorn/haelth/blob/main/doc/software_requirement_specification.md) in our repository.

## Sequence Diagrams
As with the activity diagrams, we found that [draw.io](https://app.diagrams.net/) is a great tool for creating sequence diagrams as well. We created two sequence diagrams: one visualizes the process of adding a meal to your meal plan, while the other shows today's nutrition goals.

For example:
![sequence_diagram_addMeal](https://github.com/user-attachments/assets/b86b7167-0007-4189-862f-c75eea331e5d)
 
To see all of the sequence diagrams , we advise you to see the [Software Requirement Specification](https://github.com/ldcdorn/haelth/blob/main/doc/software_requirement_specification.md) in our repository.

## User Story
Throughout this week, our team added some new user stories and included the "Done" - criteria for those that were missing it as well.

**Comment by themsgcentre** [comment]
Wir finden es gut, dass ihr eine klare Gliederung in die verschiedenen Themenbereiche vornehmt und vor allem die Aktivitätsdiagramme in Kategorien gliedert; das hilft beim Verständnis. Die Sequenzdiagramme sorgen für ein besseres Verständnis der Funktionalität eures geplanten Systems. Ebenfalls bemerkenswert finden wir, dass jeder Use-Case von einem Aktivitätsdiagramm begleitet wird, wodurch eine sehr gute Struktur und Nachvollziehbarkeit entsteht.

Uns würde eine detailliertere Version der Aktivitätsdiagramme interessieren, um die Abläufe noch besser nachvollziehen zu können. Außerdem ist uns aufgefallen, dass einige Kanten direkt mit Aktionen beschriftet sind, teils auch bei nicht bedingten Übergängen. Um die Konsistenz zu verbessern, fänden wir es sinnvoller, diese ebenfalls als Aktionsblöcke zu kennzeichnen, wie es beim klassischen Aktivitätsdiagramm der Fall ist. Dadurch werden die Diagramme klarer und verständlicher.

Bei den Sequenzdiagrammen könnte es hilfreich sein, sich intensiver mit aktiven und passiven Objekten auseinanderzusetzen, um die Interaktionen präziser darzustellen.

Wir sind sehr gespannt auf den Fortschritt der nächsten Woche!

Viele Grüße,
Projekt TDC (HM)

**Comment by MB-stack06** [comment]
An interesting post that nicely summarizes the team's progress this week! Using draw.io for creating both activity and sequence diagrams is a practical choice, especially for collaborative work. The diagrams illustrating the app's functions in Workout, Nutrition, and Cognitive Exercise categories provide clear visual structures that help in understanding the application's requirements. Adding "Done" criteria to the user stories also shows the team's commitment to quality assurance. A well-rounded insight into the development process and documented achievements. Keep up the great work!

Best regards, your GymTrackerX-team
